sql/core/src/test/scala/org/apache/spark/sql/execution/DataSourceV2ScanExecRedactionSuite.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.execution
+
+import org.apache.hadoop.fs.Path
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.QueryTest
+import org.apache.spark.sql.execution.datasources.v2.BatchScanExec
+import org.apache.spark.sql.execution.datasources.v2.orc.OrcScan
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.test.SharedSQLContext
+
+/**
+ * Suite that tests the redaction of DataSourceScanExec
+ */
+class DataSourceV2ScanExecRedactionSuite extends QueryTest with SharedSQLContext {
+
+  override protected def sparkConf: SparkConf = super.sparkConf
+    .set("spark.redaction.string.regex", "file:/[\\w_/]+")
+    .set(SQLConf.USE_V1_SOURCE_READER_LIST.key, "")
+
+  test("treeString is redacted") {
+    withTempDir { dir =>
+      val basePath = dir.getCanonicalPath
+      spark.range(0, 10).toDF("a").write.orc(new Path(basePath, "foo=1").toString)
+      val df = spark.read.orc(basePath)
+
+      val rootPath = df.queryExecution.sparkPlan.find(_.isInstanceOf[BatchScanExec]).get
+        .asInstanceOf[BatchScanExec].scan.asInstanceOf[OrcScan].fileIndex.rootPaths.head
+      assert(rootPath.toString.contains(dir.toURI.getPath.stripSuffix("/")))
+      assert(!df.queryExecution.sparkPlan.treeString(verbose = true).contains(rootPath.getName))
+      assert(!df.queryExecution.executedPlan.treeString(verbose = true).contains(rootPath.getName))
+      assert(!df.queryExecution.toString.contains(rootPath.getName))
+      assert(!df.queryExecution.simpleString.contains(rootPath.getName))
+
+      val replacement = "*********"
+      assert(df.queryExecution.sparkPlan.treeString(verbose = true).contains(replacement))
+      assert(df.queryExecution.executedPlan.treeString(verbose = true).contains(replacement))
+      assert(df.queryExecution.toString.contains(replacement))
+      assert(df.queryExecution.simpleString.contains(replacement))
+    }
+  }
+
+  private def isIncluded(queryExecution: QueryExecution, msg: String): Boolean = {
+    queryExecution.toString.contains(msg) ||
+      queryExecution.simpleString.contains(msg) ||
+      queryExecution.stringWithStats.contains(msg)
+  }
```

Comment:
We already have the same function in `DataSourceScanExecRedactionSuite`. Can we try to reduce the duplications, @cloud-fan and @gengliangwang ?
```scala
  private def isIncluded(queryExecution: QueryExecution, msg: String): Boolean = {
    queryExecution.toString.contains(msg) ||
    queryExecution.simpleString.contains(msg) ||
    queryExecution.stringWithStats.contains(msg)
  }
```

Reply:
I think this is quite minor. We should decide the principle of adding tests for data source V2:
1. Try to make tests for V2 isolated. If there is a lot of common code, create an abstract class for the test. So that someday, when the V1 is deprecated, we can remove the tests easier.

2. Put both V1 and V2 test cases together to reduce duplications.


Personally, I prefer option 1.

Reply:
We had better achieve (1) by putting the utility functions into a proper utility class or we design the class hierarchy more reasonably. Removing V1 is not hard if we have a correct design. Most design principles are for reducing duplications and reusing as much as possible.